src/main/java/org/mule/extension/ftp/internal/ftp/command/ClassicFtpCommand.java

```diff
@@ -79,7 +83,7 @@ protected void doMkDirs(Path directoryPath) {
     String cwd = getCurrentWorkingDirectory();
     Stack<Path> fragments = new Stack<>();
     try {
-      for (int i = directoryPath.getNameCount(); i >= 0; i--) {
+      for (int i = directoryPath.getNameCount(); i > 0; i--) {
```

Comment:
subpath:
    * @throws  IllegalArgumentException
     *          if {@code beginIndex} is negative, or greater than or equal to
     *          the number of elements. If {@code endIndex} is less than or
     *          equal to {@code beginIndex}, or larger than the number of elements.

we were both wrong, it can't even reach 0